### PR TITLE
Revert changes to `CODEOWNERS` that caused build failure notifications to be routed to wrong people + assorted fixes

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
Recently I've been cleaning up `CODEOWNERS` paths:

- #33597
- #33595
- #33584

And also enabled the new regex-based wildcard-supporting `CODEOWNERS` path matcher:
- https://github.com/Azure/azure-sdk-tools/pull/5241
- https://github.com/Azure/azure-sdk-tools/pull/5240

Unfortunately, this led to few issues, most notably build failure notifications being routed to wrong folks, as seen in this table:

image

I.e. for `PATH` the failures are currently routed to `RIGHT OWNERS` instead of `LEFT OWNERS`. This PR fixes that, by ensuring the failures are again routed to `LEFT OWNERS`.

This PR also restores paths for `/**/ci.yml` and `/**/tests.yml`, and does few additional fixups, including aligning all owner lists.

I am going to make a separate PR updating `iot` and other ownership information based on discussion I had over email with @jsquire and other folks.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -753,3 +783,4 @@
 
 # Client generation tooling
 /eng/templates/Azure.ServiceTemplate.Template   @chunyu3 @lirenhe
+
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:786]:
```suggestion
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -738,6 +767,7 @@
 /sdk/**/api/                                    @KrzysztofCwalina @tg-msft
 
 # ######## Eng Sys ########
+
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:770]:
```suggestion
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,6 +1,25 @@
...
+# /.devcontainer/
+# /.vscode/
+# /common/Perf/
+# /common/Stress/
+# /doc/
+# /samples/
+# /tools/
+
+# Last chance Eng Sys catch-all
+/**/tests.yml   @hallipr @benbp
```
Comment (weshaggard) [inline .github/CODEOWNERS:20]: Lets keep these removed.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -111,6 +130,7 @@
 
 # ServiceLabel: %Azure Stack %Service Attention
 /sdk/azurestack/            @sijuman @sarathys @bganapa @rakku-ms
+/sdk/azurestackhci/*.yml    @sijuman @sarathys @bganapa @rakku-ms
```
Comment (weshaggard) [inline .github/CODEOWNERS:133]: Why only *.yml? I would consider `/sdk/azurestack*`

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -673,6 +697,11 @@
 # PRLabel: %Storage
 /sdk/storage/Microsoft.Azure.WebJobs.*/         @seanmcc-msft @amnguye @jaschrep-msft @tg-msft @fabiocav @mathewc
 
+/sdk/storagecache/*.yml                         @seanmcc-msft @amnguye @jaschrep-msft @tg-msft
```
Comment (weshaggard) [inline .github/CODEOWNERS:700]: Should we do `/sdk/storage*` above?

Comment (konrad-jamrozik) [reply .github/CODEOWNERS:700]: @ArthurMa1978:

Should I assign the same set of owners to all these directories, in `/sdk` ?
image

Per @jsquire:

> To my knowledge, the Storage team owns all storage client libraries in our repository. If those paths are management libraries, which I’m not sure is the case, those should go to @ArthurMa1978 (...) who own the management plane all-up.

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -1,755 +1,773 @@
...
-/sdk/communication/Azure.Communication.ShortCodes/      @guilhermeluizsp @danielav7
+/sdk/communication/Azure.Communication.ShortCodes/                 @guilhermeluizsp @danielav7
 
 # PRLabel: %Communication - SMS
-/sdk/communication/Azure.Communication.Sms/             @RoyHerrod @arifibrahim4
+/sdk/communication/Azure.Communication.Sms/                        @RoyHerrod @arifibrahim4
 
+# TODO: OWNERS MISSING HERE!
 # PRLabel: %Communication - Resource Manager
 /sdk/communication/Azure.ResourceManager.Communication/
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:256]: @jsquire FYI owners missing here. See:
- https://github.com/Azure/azure-sdk-for-net/pull/27760

Comment (jsquire) [reply .github/CODEOWNERS:256]: That's a management library.    It's Arthur Ma and Elendil Zheng.

Comment (jsquire) [reply .github/CODEOWNERS:256]: All. *.ResourceManager.* are those folks.

Comment (konrad-jamrozik) [reply .github/CODEOWNERS:256]: OK, this path is superseded by `/**/Azure.ResourceManager*/` below anyway. I am thus going to remove it.

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -1,755 +1,773 @@
...
-#/<NotInRepo>/                                   @xgithubtriage
+#/<NotInRepo>/                                                     @xgithubtriage
 
 # ServiceLabel: %Storsimple %Service Attention
-/sdk/storsimple/Microsoft.Azure.Management.StorSimple/           @anoobbacker @ganzee @manuaery @patelkunal
+/sdk/storsimple/Microsoft.Azure.Management.StorSimple/             @anoobbacker @ganzee @manuaery @patelkunal
 
+# TODO: OWNERS MISSING HERE!
 # ServiceLabel: %Storsimple %Service Attention
 /sdk/storsimple8000series/Microsoft.Azure.Management.StorSimple8000Series/
```
[the remark this answers is not in the document]
Comment (konrad-jamrozik) [reply .github/CODEOWNERS:701]: I am going to assign folks for which it currently matches, which is:
`https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS#L734`

**Review by weshaggard — APPROVED** [review]

**Pull request merged**